js/src/ui/TxList/store.js

```diff
@@ -14,150 +14,121 @@
-    this._api
-      .subscribe('eth_blockNumber', (error, blockNumber) => {
-        if (error) {
-          return;
-        }
-
-        if (this._pendingHashes.length) {
-          this.loadTransactions(this._pendingHashes);
-          this.clearPending();
-        }
-      })
-      .then((subscriptionId) => {
-        this._subscriptionId = subscriptionId;
-      });
+  sameHashList = (transactions) => {
+    return this.containsAll(transactions, this.sortedHashes) && this.containsAll(this.sortedHashes, transactions);
   }
 
-  unsubscribe () {
-    if (!this._subscriptionId) {
-      return;
-    }
+  sortHashes = () => {
+    this.sortedHashes = this.sortedHashes.sort((hashA, hashB) => {
+      const bnA = this.transactions[hashA].blockNumber;
+      const bnB = this.transactions[hashB].blockNumber;
 
-    this._api.unsubscribe(this._subscriptionId);
-    this._subscriptionId = 0;
+      // 0 is a special case (has not been added to the blockchain yet)
+      if (bnB.eq(0)) {
+        return bnB.eq(bnA) ? 0 : 1;
+      } else if (bnA.eq(0)) {
+        return -1;
+      }
+
+      return bnB.comparedTo(bnA);
+    });
   }
 
-  loadTransactions (_txhashes = []) {
-    const promises = _txhashes
-      .filter((txhash) => !this.transactions[txhash] || this._pendingHashes.includes(txhash))
-      .map((txhash) => {
-        return Promise
-          .all([
-            this._api.eth.getTransactionByHash(txhash),
-            this._api.eth.getTransactionReceipt(txhash)
-          ])
-          .then(([
-            transaction = {},
-            transactionReceipt = {}
-          ]) => {
-            return {
-              ...transactionReceipt,
-              ...transaction
-            };
-          });
-      });
+  loadTransactions (_txhashes) {
+    const { eth } = this._api;
 
-    if (!promises.length) {
+    // Ignore special cases and if the contents of _txhashes && this.sortedHashes are the same
+    if (Array.isArray(_txhashes) || this.sameHashList(_txhashes)) {
+      return;
+    }
+    // If there was a hash removed:
+    if (_txhashes.length < this.sortedHashes.length) {
+      this.removeHash(_txhashes);
       return;
     }
 
-    Promise
-      .all(promises)
-      .then((_transactions) => {
-        const blockNumbers = [];
-        const transactions = _transactions
-          .filter((tx) => tx && tx.hash)
-          .reduce((txs, tx) => {
-            txs[tx.hash] = tx;
-
-            if (tx.blockNumber && tx.blockNumber.gt(0)) {
-              blockNumbers.push(tx.blockNumber.toNumber());
+    _txhashes
+      .map((txhash) => {
+        eth.getTransactionByHash(txhash)
+          .then((tx) => {
+            this.transactions[txhash] = tx;
+            // If the tx has a blockHash, let's get the blockNumber, otherwise it's ready to be added
+            if (tx.blockHash) {
+              eth.getBlockByNumber(tx.blockNumber)
+                .then((block) => {
+                  this.blocks[tx.blockNumber] = block;
```

Comment:
Yes, but if I leave my page open on transaction list it will consume all memory after some time. IMHO its an obvious memory leak and we shouldn't tell our users to just refresh the page from time to time.